pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -287,3 +288,35 @@ func (s *NSXServiceAccountService) GetNSXServiceAccountNameByUID(uid string) (na
 func (s *NSXServiceAccountService) getClusterName(namespace, name string) string {
 	return fmt.Sprintf("%s-%s-%s", s.NSXConfig.CoeConfig.Cluster, namespace, name)
 }
+
+func GenerateNSXServiceAccountConditions(existingConditions []metav1.Condition, generation int64, realizedStatus metav1.ConditionStatus, reason string) []metav1.Condition {
+	var conditions []metav1.Condition
+	for i := range existingConditions {
+		switch existingConditions[i].Type {
+		case v1alpha1.ConditionTypeRealized:
+		default:
+			conditions = append(conditions, *existingConditions[i].DeepCopy())
+		}
+	}
+	conditions = append(conditions, metav1.Condition{
+		Type:               v1alpha1.ConditionTypeRealized,
+		Status:             realizedStatus,
+		ObservedGeneration: generation,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            "",
```

Comment:
Message is not used here, Message is a human readable message indicating details about the transition. This may be an empty string. What if condition is false, will Message be empty?

Reply:
Yes, unless we want to design a human reading message for this case.

Reply:
Human readable message will be helpful in debugging realization failure. If it's possible, we can put the direct error message which causes the failure here.

Reply:
Currently we put the error to reason, thus it is duplicated if we put it to message.